src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQQueueClient.cs

```diff
@@ -144,53 +141,20 @@ public RabbitMQQueueClient(ConnectionFactory connectionFactory, string queueName
             Func<IQueueMessage, CancellationToken, Task> messageHandler,
             MessageHandlerOptions messageHandlerOptions)
         {
+            // Reconfigure QoS in order to support specified level of concurrency.
+            ConfigureBasicQos(prefetchCount: messageHandlerOptions.MaxConcurrentCalls);
```

Comment:
It is a shame to have to call invoke `_channel.BasicQos` every time `RegisterMessageHandler` is called. I couldn't see any alternatives given the current API design.

Reply:
I think this is fine. Note that other queue client implementations only permit a single message handler to be registered (for a given instance of the queue client). We should copy over this behaviour to the RabbitMQ client too. If making this change, then we'd only ever call `_channel.BasicQos` (up to) twice per instance of the queue client.